packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -679,10 +715,29 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
         const pendingRuntimeState = context.pendingLocalState as IPendingRuntimeState | undefined;
         const baseSnapshot: ISnapshotTree | undefined = pendingRuntimeState?.baseSnapshot ?? context.baseSnapshot;
-        const storage = !pendingRuntimeState ?
+        const summaryConfiguration: ISummaryConfigurationHeuristics =
+            ContainerRuntime.buildSummaryConfiguration(runtimeOptions) as ISummaryConfigurationHeuristics;
+        let storage = !pendingRuntimeState ?
             context.storage :
             new SerializedSnapshotStorage(() => { return context.storage; }, pendingRuntimeState.snapshotBlobs);
+        if (summaryConfiguration?.compressionAlgorithm !== undefined) {
+            storage = buildSummaryStorageAdapter(storage,
+                [new CompressionSummaryStorageHooks(
+                    summaryConfiguration.compressionAlgorithm,
+                    summaryConfiguration.minSizeToCompress,
+                    summaryConfiguration.isUseB64OnCompressed)],
+            );
+            /*
```

Comment:
@vladsud, @anthony-murphy - Would you recommend using 'this.logger' to log when compression is enabled?

If not, is DebugLogger acceptable?  (I ask because I don't see much use of DebugLogger outside of tests.)

Reply:
@milanro - Probably the best approach is to record stats about compression into the telemetry context during summarization (e.g., algorithm used, num blobs compressed, avg. compression ratio.)

Here is where the telemetry context is created in `ContainerRuntime.summarize`.  When summarize returns, the properties in the context are serialized into the 'SummarizeTelemetry' event.

```ts
        const telemetryContext = new TelemetryContext();
        const { stats, summary } = await this.summarizerNode.summarize(fullTree, trackState, telemetryContext);

        this.logger.sendTelemetryEvent({ eventName: "SummarizeTelemetry", details: telemetryContext.serialize() });
```

Set `localStorage.debug = 'fluid:*'` to view the output (and make sure your console is not filtering out verbose events):
https://fluidframework.com/docs/testing/telemetry/

(You can also refine the filter to 'fluid:telemetry:Summarizer:*' to view just the logs for the summarizer.)